datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -1054,8 +1118,8 @@ impl DependencyNode {
     }
 }
 
-type DependencyMap = HashMap<PhysicalSortExpr, DependencyNode>;
-type Dependencies = HashSet<PhysicalSortExpr>;
+type DependencyMap = IndexMap<PhysicalSortExpr, DependencyNode>;
```

Comment:
Maybe we can add a comment or two explaining *why* this uses an IndexMap and not a HashMap presumably because the order matters? Or it is required to get consistent plans?